packages/select/src/components/query-list/queryList.tsx

```diff
@@ -350,6 +369,40 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
         }
     };
 
+    private handlePaste = (values: string[]) => {
+        const { createNewItemFromQuery, onItemsPaste } = this.props;
+
+        let nextActiveItem: T | undefined;
+
+        // Find an exising item that exactly matches each pasted value, or
+        // create a new item if possible. Ignore unmatched values if creating
+        // items is disabled.
+        const pastedItemsToEmit = values.reduce<T[]>((agg, value) => {
+            const equalItem = getMatchingItem(value, this.props);
+
+            if (equalItem !== undefined) {
+                nextActiveItem = equalItem;
+                agg.push(equalItem);
+            } else if (this.canCreateItems()) {
+                const newItem = Utils.safeInvoke(createNewItemFromQuery, value);
+                if (newItem !== undefined) {
+                    agg.push(newItem);
+                }
+            }
+
+            return agg;
+        }, []);
+
+        // UX nicety: update the active item if we matched with at least one
+        // existing item.
+        if (nextActiveItem !== undefined) {
+            this.setActiveItem(nextActiveItem);
+        }
+
+        // No need to update the active item.
```

Comment:
what's this comment about? you _may_ have updated the active item at this point, no?

Reply:
Old cruft. Removed. 👍